app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
```

Comment:
dấu ||= để làm gì?

Reply:
Hiểu đơn giản thì nó  là toán tử dùng để gán giá trị:
 a ||= b
Nếu giá trị của a là nil thì câu lênh trên sẽ gán giá trị cho a = b, nếu giá trị của a khác nil thì gía trị của a được giữ nguyên

Reply:
Trong trường hợp này là do tính tái sử dụng của biến instance trong cùng một request/respond, nếu sử dụng là biến local thì mỗi lần gọi đến current_user thì luôn phải tìm trong database để gán giá trị cho biến. Trong trường hợp nhiều request sẽ ảnh hưởng đến performance, còn biến instance việc này chỉ thực hiện một lần, giá trị sẽ được lưu lại và tái sử dụng.